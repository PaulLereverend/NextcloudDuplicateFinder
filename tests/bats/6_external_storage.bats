#!/usr/bin/env bats
TESTSUITE="duplicatefinder"
load helper.sh

setup() {
  if [ -z "$STORAGE_HOST" ]; then
    echo "environment variable STORAGE_HOST need to bee defined"
    exit 1
  fi
  DO_NOT_SCAN=1
  load ${BATS_TEST_DIRNAME}/setup.sh
  ./occ app:enable files_external
  sed "s/%%host%%/$STORAGE_HOST/g" ${BATS_TEST_DIRNAME}/assets/externalStorage.json > externalStorage.json
  for m in $(./occ files_external:list | grep -E '^\| [1-9]' | awk '{print $2}'); do
    ./occ files_external:delete -y $m
  done
  ./occ files_external:import externalStorage.json
  find ./data/admin/files/tests -type f -name '2_*' \
    -exec curl -sS -u 'test:test' -T {} http://$STORAGE_HOST/webdav/ \; \
    -exec curl -sS -u 'test:test' -T {} ftp://$STORAGE_HOST \; \
    -exec curl -sS -u 'WORKGROUP\test:test' -T {} smb://$STORAGE_HOST/public/ \; >/dev/null
}

teardown() {
    ./occ -v duplicates:clear -f
    for m in $(./occ files_external:list | grep -E '^\| [1-9]' | awk '{print $2}'); do
      ./occ files_external:delete -y $m
    done
    clearTestFiles
}

@test "[$TESTSUITE] Duplicates on external storage" {
    ./occ config:app:delete duplicatefinder ignore_mounted_files
    
    output=$(./occ -v duplicates:find-all -u admin)
    expectedHash="2ba3a0e386a9784998f74b45a542871db0d84dccb708c69c908d408b74fafbab"
    evaluateHashResult "${expectedHash}" 3 "${output}"
}

@test "[$TESTSUITE] Skip search for duplicates on external storage" {
    ./occ config:app:set --value=true duplicatefinder ignore_mounted_files
    
    output=$(./occ -v duplicates:find-all -u admin)
    expectedHash="cbdc41d9d24253c54da8cd3087ecde71b45c7c9f328336d1247d2032b3a5aa28"
    evaluateHashResult "${expectedHash}" 3 "${output}"
}
