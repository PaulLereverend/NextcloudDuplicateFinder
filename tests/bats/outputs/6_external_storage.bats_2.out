Start searching files for admin in path /admin/files
Scanning /admin/files/tests/3_3.txt
Scanning /admin/files/tests/0_0.txt
Scanning /admin/files/tests/3_2.txt
Scanning /admin/files/tests/1_1.txt
Scanning /admin/files/tests/2_0.txt
Scanning /admin/files/tests/1_0.txt
Scanning /admin/files/tests/3_1.txt
Scanning /admin/files/tests/2_2.txt
Scanning /admin/files/tests/3_0.txt
Scanning /admin/files/tests/2_1.txt
Scanning /admin/files/eWebDAV/2_0.txt
Skipped /admin/files/eWebDAV/2_0.txt
Scanning /admin/files/eWebDAV/2_2.txt
Skipped /admin/files/eWebDAV/2_2.txt
Scanning /admin/files/eWebDAV/2_1.txt
Skipped /admin/files/eWebDAV/2_1.txt
Scanning /admin/files/eFTP/2_0.txt
Skipped /admin/files/eFTP/2_0.txt
Scanning /admin/files/eFTP/2_2.txt
Skipped /admin/files/eFTP/2_2.txt
Scanning /admin/files/eFTP/2_1.txt
Skipped /admin/files/eFTP/2_1.txt
Scanning /admin/files/eSMB/2_0.txt
Skipped /admin/files/eSMB/2_0.txt
Scanning /admin/files/eSMB/2_2.txt
Skipped /admin/files/eSMB/2_2.txt
Scanning /admin/files/eSMB/2_1.txt
Skipped /admin/files/eSMB/2_1.txt
Finished searching files
Duplicates for user "admin" are: 
1121cfccd5913f0a63fec40a6ffd44ea64f9dc135c66634ba001d10bcf4302a2(file_hash)
     /admin/files/tests/3_0.txt
     /admin/files/tests/3_1.txt
     /admin/files/tests/3_2.txt
     /admin/files/tests/3_3.txt
4355a46b19d348dc2f57c046f8ef63d4538ebb936000f3c9ee954a27460dd865(file_hash)
     /admin/files/tests/1_0.txt
     /admin/files/tests/1_1.txt
53c234e5e8472b6ac51c1ae1cab3fe06fad053beb8ebfd8977b010655bfdd3c3(file_hash)
     /admin/files/tests/2_0.txt
     /admin/files/tests/2_1.txt
     /admin/files/tests/2_2.txt
